File: android-lib/src/androidTest/java/com/kinvey/androidTest/store/user/DefaultKinveyVoidCallback.java
package com.kinvey.androidTest.store.user;

import com.kinvey.java.core.KinveyClientCallback;

import java.util.concurrent.CountDownLatch;

public class DefaultKinveyVoidCallback implements KinveyClientCallback<Void> {

    private CountDownLatch latch;
    Void result;
    Throwable error;

    DefaultKinveyVoidCallback(CountDownLatch latch) {
        this.latch = latch;
    }

    @Override
    public void onSuccess(Void result) {
        this.result = result;
        finish();
    }

    @Override
    public void onFailure(Throwable error) {
        this.error = error;
        finish();
    }

    private void finish() {
        latch.countDown();
    }
}

File: android-lib/src/androidTest/java/com/kinvey/androidTest/store/user/DefaultKinveyVoidCallback.kt
package com.kinvey.androidTest.store.user

import com.kinvey.java.core.KinveyClientCallback
import java.util.concurrent.CountDownLatch

class DefaultKinveyVoidCallback(private val latch: CountDownLatch) : KinveyClientCallback<Void> {

    var result: Void? = null
    var error: Throwable? = null

    override fun onSuccess(result: Void?) {
        this.result = result
        finish()
    }

    override fun onFailure(error: Throwable?) {
        this.error = error
        finish()
    }

    private fun finish() {
        latch.countDown()
    }
}